chain-core/azuredeploy.json

```diff
@@ -0,0 +1,244 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "defaultValue": null,
+            "metadata": {
+                "description": "Adminstrator user name"
+            },
+            "type": "String"
+        },
+        "adminPassword": {
+            "defaultValue": null,
+            "metadata": {
+                "description": "Adminstrator password"
+            },
+            "type": "SecureString"
+        },
+        "coreVmName": {
+            "defaultValue": "",
+            "metadata": {
+                "description": "Virtual machine name"
+            },
+            "type": "String"
+        },
+        "virtual_machine_size": {
+            "defaultValue": "Standard_D2",
+            "metadata": {
+                "description": "Size of virtual machine"
+            },
+            "type": "String",
+            "allowedValues": [
+                "Standard_D1",
+                "Standard_D2",
+                "Standard_D3",
+                "Standard_D4",
+                "Standard_DS1",
+                "Standard_DS2",
+                "Standard_DS3",
+                "Standard_DS4"
+            ]
+        }
+    },
+    "variables": {
+        "location": "[resourceGroup().location]",
+        "vmSize": "Standard_D2",
+        "storageAccount": "[parameters('coreVmName')]",
+        "nicName": "[parameters('coreVmName')]",
+        "publicIpName": "[parameters('coreVmName')]",
+        "vnetName": "[parameters('coreVmName')]",
+        "nsgName": "[parameters('coreVmName')]",
+        "dnsName": "[parameters('coreVmName')]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[parameters('coreVmName')]",
+            "apiVersion": "2015-06-15",
+            "location": "[variables('location')]",
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('virtual_machine_size')]"
+                },
+                "storageProfile": {
+                    "imageReference": {
+                        "publisher": "Canonical",
+                        "offer": "UbuntuServer",
+                        "sku": "16.04.0-LTS",
+                        "version": "latest"
+                    },
+                    "osDisk": {
+                        "name": "[concat(parameters('coreVmName'), 'os_disk')]",
+                        "createOption": "FromImage",
+                        "vhd": {
+                            "uri": "[concat('https', '://', variables('storageAccount'), '.blob.core.windows.net', concat('/vhds/', variables('storageAccount'),'.vhd'))]"

```

Comment:
please see the "all clouds" best practice: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md